centreon/tests/e2e/features/Dashboards/10-widget-metric-graph-configuration/index.ts

```diff
@@ -0,0 +1,398 @@
+  }).as('performanceData');
+  cy.loginByTypeOfUser({
+    jsonName: dashboardAdministratorUser.login,
+    loginViaApi: true
+  });
+  cy.logoutViaAPI();
+});
+
+beforeEach(() => {
+  cy.intercept({
+    method: 'GET',
+    url: '/centreon/api/internal.php?object=centreon_topology&action=navigationList'
+  }).as('getNavigationList');
+  cy.intercept({
+    method: 'GET',
+    url: '/centreon/api/latest/configuration/dashboards?'
+  }).as('listAllDashboards');
+  cy.intercept({
+    method: 'POST',
+    url: `/centreon/api/latest/configuration/dashboards/*/access_rights/contacts`
+  }).as('addContactToDashboardShareList');
+  cy.intercept({
+    method: 'PATCH',
+    url: `/centreon/api/latest/configuration/dashboards/*`
+  }).as('updateDashboard');
+  cy.intercept({
+    method: 'GET',
+    url: /\/api\/latest\/monitoring\/dashboard\/metrics\/performances\/data\?.*$/
+  }).as('performanceData');
+  cy.loginByTypeOfUser({
+    jsonName: dashboardAdministratorUser.login,
+    loginViaApi: false
+  });
+  cy.visit('/centreon/home/dashboards');
+});
+
+afterEach(() => {
+  cy.requestOnDatabase({
+    database: 'centreon',
+    query: 'DELETE FROM dashboard'
+  });
+});
+
+after(() => {
+  cy.stopWebContainer();
+});
+
+Given(
+  "a dashboard in the dashboard administrator user's dashboard library",
+  () => {
+    cy.insertDashboard({ ...dashboards.default });
+    cy.visit(`${Cypress.config().baseUrl}/centreon/home/dashboards`);
+    cy.getByLabel({
+      label: 'view',
+      tag: 'button'
+    })
+      .contains(dashboards.default.name)
+      .click();
+  }
+);
+
+When(
+  'the dashboard administrator user selects the option to add a new widget',
+  () => {
+    cy.get('*[class^="react-grid-layout"]').children().should('have.length', 0);
+    cy.getByTestId({ testId: 'edit_dashboard' }).click();
+    cy.getByTestId({ testId: 'AddIcon' }).click();
+  }
+);
+
+When('selects the widget type "Metrics graph"', () => {
+  cy.getByTestId({ testId: 'Widget type' }).click();
+  cy.contains('Metrics graph').click();
+});
+
+Then(
+  'configuration properties for the Metrics graph widget are displayed',
+  () => {
+    cy.contains('Widget properties').should('exist');
+    cy.getByLabel({ label: 'Title' }).should('exist');
+    cy.getByLabel({ label: 'RichTextEditor' }).should('exist');
+    cy.getByTestId({ testId: 'Time period' }).should('exist');
+  }
+);
+
+When(
+  'the dashboard administrator user selects a resource and a metric for the widget to report on',
+  () => {
+    cy.getByLabel({ label: 'Title' }).type(genericTextWidgets.default.title);
+    cy.getByLabel({ label: 'RichTextEditor' })
+      .eq(0)
+      .type(genericTextWidgets.default.description);
+    cy.getByTestId({ testId: 'Resource type' }).realClick();
+    cy.getByLabel({ label: 'Host Group' }).click();
+    cy.getByTestId({ testId: 'Select resource' }).click();
+    cy.get('[class^="MuiAutocomplete-listbox"]').click();
```

Comment:
is it the action to select a resource ?
could you select a resource by its name to make the test more readable